pre-commit

```diff
@@ -77,25 +77,28 @@ check() {
 }
 
 check "Benchmark compilation" \
-    cargo build --benches --no-default-features --features "enable-benches"
+    cargo build -p ipa --benches --no-default-features --features "enable-benches"
 
 check "Clippy checks" \
     cargo clippy --tests -- -D warnings
 
 check "Clippy concurrency checks" \
-    cargo clippy --tests --features shuttle -- -D warnings
+    cargo clippy -p ipa --tests --features="shuttle in-memory-infra" -- -D warnings
 
 check "Tests" \
     cargo test
 
+check "IPA Tests" \
+    cargo test -p ipa --features in-memory-infra
+
 check "Web tests" \
     cargo test --no-default-features --features "cli web-app real-world-infra test-fixture"
```

Comment:
q: does it run both ipa and ipa-helper tests?

Reply:
This one only runs ipa tests. ipa-helper tests are run in "Web tests".

With "IPA tests", pre-commit covers all unit tests, but there are some overwrap in "Tests" and "IPA Tests". For example, unit tests in `ipa/src/exact.rs` have `#[cfg(all(test, not(feature = "shuttle")))]` which is true in both cases. We could add `not(in-memory-infra)` if we want to avoid the overwrap, but seems a hacky way of fixing this.